pulp_container/app/registry_api.py

```diff
@@ -355,6 +356,7 @@ class BearerTokenView(APIView):
     """
 
     # Allow everyone to access but still value authenticated users.
+    authentication_classes = [BasicAuthentication]
```

Comment:
Does it still allow handing out anonymous tokens?

Reply:
Yes:
```
(pulp) [vagrant@pulp3-source-fedora35 ~]$ http -vv https://pulp3-source-fedora35.localhost.example.com/token/?service="localhost:24817"
GET /token/?service=localhost:24817 HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: pulp3-source-fedora35.localhost.example.com
User-Agent: HTTPie/3.2.1



HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 1077
Content-Type: application/json
Correlation-ID: a82a1399d7c043b498a13efd94fa6af2
Date: Thu, 18 Aug 2022 06:55:40 GMT
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJsb2NhbGhvc3Q6MjQ4MTciLCJleHAiOjE2NjA4MDYwNDAsImlhdCI6MTY2MDgwNTc0MCwiaXNzIjoiaHR0cHM6Ly9wdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tL3Rva2VuLyIsImp0aSI6ImEyNDFiZTNiLTQ3NTgtNDA5NC1iNTQ4LWJmYzU2ZjAxMjU0MiIsIm5iZiI6MTY2MDgwNTc0MCwic3ViIjoiIn0.3FQ7QLLAzuMEPhbVfocPyyvaiRk9Eifus_NHDWgaB3-NDYBH1n57AIE95mDPhMayK5kKpG41DolG4_Wyvt1azw",
    "expires_in": 300,
    "issued_at": "2022-08-18T06:55:40.471912Z",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJsb2NhbGhvc3Q6MjQ4MTciLCJleHAiOjE2NjA4MDYwNDAsImlhdCI6MTY2MDgwNTc0MCwiaXNzIjoiaHR0cHM6Ly9wdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tL3Rva2VuLyIsImp0aSI6ImEyNDFiZTNiLTQ3NTgtNDA5NC1iNTQ4LWJmYzU2ZjAxMjU0MiIsIm5iZiI6MTY2MDgwNTc0MCwic3ViIjoiIn0.3FQ7QLLAzuMEPhbVfocPyyvaiRk9Eifus_NHDWgaB3-NDYBH1n57AIE95mDPhMayK5kKpG41DolG4_Wyvt1azw"
}
```

Reply:
Actually, give me a second to test this thing properly.

Reply:
```
(pulp) [vagrant@pulp3-source-fedora35 ~]$ http -vv https://pulp3-source-fedora35.localhost.example.com/token/?service="pulp3-source-fedora35.localhost.example.com"
GET /token/?service=pulp3-source-fedora35.localhost.example.com HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: pulp3-source-fedora35.localhost.example.com
User-Agent: HTTPie/3.2.1



HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 1151
Content-Type: application/json
Correlation-ID: b251aec2b55a4d06aa45a885f5173a81
Date: Thu, 18 Aug 2022 07:10:17 GMT
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ",
    "expires_in": 300,
    "issued_at": "2022-08-18T07:10:17.004930Z",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ"
}


Elapsed time: 0.025852835s

(pulp) [vagrant@pulp3-source-fedora35 ~]$ http https://pulp3-source-fedora35.localhost.example.com/v2/ --auth-type=jwt --auth=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 2
Content-Type: application/json
Correlation-ID: 36537c439271499d887520dfda972206
Date: Thu, 18 Aug 2022 07:10:26 GMT
Docker-Distribution-Api-Version: registry/2.0
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Registry-Supports-Signatures: 1

{}
```